### DISCUSSION TITLE
Classify Asyncio/Bleak

### DISCUSSION THREAD

**Discussion by andrewgrabbs** [hbldh/bleak#1312] [Q&A] [answered]
Hello! 
I was wondering if it is possible to make a single class out of your bleak library, I saw the queue example, but it still requires a main loop running. I'm trying to come up with a with a way to make something like this work...

```
driver = BleDriver(address)
driver.connect()
driver.subscribe(CHAR)
driver.read_char(CHAR)
driver.write_char(CHAR, data)
driver.unsubscribe(CHAR)
driver.disconnect()
```

Ideally I'd like to be able to process data while the connection is kept open and make decisions on what to do next without having to define the entire process before running the loop.

**Comment by dlech** [comment, marked as answer]
If you don't want asyncio, try https://pypi.org/project/simplepyble/.

If you just want an async REPL with top-level await, run `python -m asyncio`.

**Reply by andrewgrabbs** [reply]
Thanks!
